### PR TITLE
Connect Bookstore to API

### PR DESCRIPTION
**Pull request by tomdan-ai**
### **In This Milestone:**

- [ ] Connect React app to [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739)
- [ ] Use [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make your HTTP requests to the backend
- [ ] Add Loading state
- [ ] Return error if API fails

### CODE REVIEW

File: src/components/BookList.js [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
+
+  const bookComponents = [];
+
+  for (let i = 0; i < books.length; i += 1) {
+    const book = books[i];
+    bookComponents.push(<IndividualBook key={book.item_id} book={book} onDelete={onDelete} />);
+  }
+
+  return <div className={style.bookList}>{bookComponents}</div>;
+};
```
Comment (brainconnect93) [inline src/components/BookList.js]: - [ ] Hello @tomdan-ai, great effort working on the connect api milestone of the bookstore project. According to the requirement of this project, it was stated that user should be able to add and delete books. But currently, i noticed i could not add any book and whenever i try to add a book, it throw an error saying `"Error occurred while fetching books."`

Image 👇 

image

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
### Hi @tomdan-ai, 👋 

# STATUS: Changes Required :recycle:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To Highlight 👏 

-  Descriptive Pull Request title and summary. ✔️
- `CreateAsyncThunk` is imported in the project ✔️
- `Axios` or `fetch` is used to make the HTTP request ✔️
- `getBooks` actually updates the state of `booksSlice` ✔️
- `useEffect` is used when dispatching the `getBooks` function in `<BookList>` ✔️
- The `addBook` and `removeBook` features works ✔️
- Good use of Git flow ✔️

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -1,17 +1,94 @@
...
+
+    addBook: (state, { payload }) => {
+      state.books.push(payload);
     },
-    removeBook: (state, action) => state.filter((book) => book.item_id !== action.payload),
+
+    removeBook: (state, { payload }) => {
+      state.books = state.books.filter((book) => book.item_id !== payload);
+    },
+
```
Comment (Reem-lab) [inline src/redux/books/booksSlice.js]: - [ ] Great job working on the required changes so far but I think you still have a small problem when I try to delete a book so whenever I remove a book from the list I could see allthe books are removed as well but I think it shouldn't be like that so please fix that problem to just whenever the user want to remove a specific book it removed the book  not all list 🔥💯⭐🚀


https://github.com/tomdan-ai/Bookstore/assets/58553711/7c2f17e9-be5c-47d7-a07b-2a09808fe511

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi @tomdan-ai, 👋🏻👋🏻👋🏻

Great work on making the changes requested by a previous reviewer 👏 🚀💯🔥
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- NO Linters error 🔥
- Followed Gitflow 💯
- Great working on fetching the books 💪🏻

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ibizugbe — APPROVED** [review]
Hi @tomdan-ai 👋👋👋, 

StandingOvationGIF (7)


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅

- [x] Linters are passing👍.
- [x] It is possible to remove a specific book👍.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**